src/settings.js

```diff
@@ -2,44 +2,76 @@ import { isPlainObject } from './helpers';
 import { Value } from './values';
 
 export default class Settings {
-  static prefix;
-
-  constructor(settings) {
-    this.cache = new Map();
-    this.proxy = new Proxy(settings, {
-      get: (target, key) => {
-        if (this.cache.has(key)) {
-          return this.cache.get(key);
-        }
+  static assign(...args) {
+    return new Settings(
+      Object.assign(
+        ...args.map(arg => {
+          if (arg instanceof Settings) {
+            const settings = arg._settings;
+            Object.entries(settings).forEach(([key, value]) => {
+              if (value instanceof Value && value.options.envPrefix === undefined) {
+                settings[key].options.envPrefix = arg._prefix;
+              }
+            });
+            return settings;
+          }
+          return arg;
+        })
+      )
+    );
+  }
 
-        const value = this.parseValue(target[key], key);
-        this.cache.set(key, value);
-        return value;
+  constructor(settings = {}, prefix) {
+    const cache = new Map();
+
+    // We use defineProperties because we don't want these to be enumerable
```

Comment:
I would set `enumerable: false` explicitly, because I can't remember what the default is.